src/main/scala/is/hail/utils/richUtils/RichDenseMatrixDouble.scala

```diff
@@ -19,17 +20,17 @@ object RichDenseMatrixDouble {
     val rows = in.readInt()
     val cols = in.readInt()
     val isTranspose = in.readBoolean()
-    val majorStride = if (isTranspose) cols else rows
+    val length = rows * cols
     
-    val data = new Array[Double](rows * cols)
-    var i = 0
-    while (i < rows * cols) {
-      data(i) = in.readDouble()
-      i += 1
-    }
+    val n = length << 3
+    val bytes = new Array[Byte](n)
+    in.read(bytes, 0, n)
```

Comment:
This needs to be `readFully`.  This is probably your bug.

Reply:
Ha, I started writing a loop for reading fully but your note that it exists. I think the issue may be that writeDouble on DataOutputStream first converts to a long representation using doubleToLongBits, which may not be the raw bytes. I am going to change the IRM to Block export to use Array[Byte] as well.

I also realized that using an Array[Byte] divides by 8 the largest dimension of local matrix we can output, from 46k to 5792. Unless you have a better idea, I'll fix that by putting the write in a loop that takes up to maxInt bytes at a time.